tiled/client/catalog.py

```diff
@@ -21,6 +23,26 @@
 from .utils import handle_error
 
 
+def get_token(uri):
+    username = input("Username: ")
+    password = getpass.getpass()
+    form_data = {'grant_type': 'password',
+                 'username': username,
+                 'password': password}
+    response = httpx.post(uri + '/token', data=form_data)
+    return 'Bearer ' + json.loads(response.text)['access_token']
+
+
+def get_public_token(uri):
```

Comment:
Interesting idea. Placing something in the `Authorization` header that is actually not a secret but in fact a hard-coded public constant feels off to me. What would it look like to make a special case in the server where, if Authorization is not set, the user is treated as `SpecialUsers.public`?

Another consideration is that any string we use for username here could potentially collide with an actual username in whatever user management system we integrate with. (Perhaps some guy Paul Ublic has public@bnl.gov — who knows!) It’s unlikely but it is cleaner to make the public “user” a special sentinel that is “out of band” from the username namespace, as I have done in the sketch in `main`.